File: apps/mobile/lib/features/transactions/data/dtos/get_transaction_response_dto.dart
import 'package:json_annotation/json_annotation.dart';

part 'get_transaction_response_dto.g.dart';

@JsonSerializable(createToJson: false)
class GetTransactionResponseDto {
  const GetTransactionResponseDto({
    required this.id,
    required this.subCategoryId,
    required this.subCategoryName,
    required this.accountId,
    required this.accountName,
    required this.value,
    required this.type,
    required this.transactionDate,
    required this.paymentType,
    this.budgetId,
    this.description,
    this.recurringTransactionId,
    this.parentTransactionId,
    this.installmentNumber,
    this.totalInstallments,
    this.areaId,
    this.areaName,
    this.paymentMethod,
  });

  final int id;
  final int subCategoryId;
  final String subCategoryName;
  final int accountId;
  final String accountName;

  /// In cents. R$ 150,00 = 15000.
  final int value;

  /// "Expense" | "Income"
  final String type;

  /// Format: "YYYY-MM-DD"
  final String transactionDate;

  /// "OneTime" | "Installment" | "Recurring"
  final String paymentType;

  final int? budgetId;
  final String? description;
  final int? recurringTransactionId;
  final int? parentTransactionId;
  final int? installmentNumber;
  final int? totalInstallments;
  final int? areaId;
  final String? areaName;

  /// "Credit" | "Debit" | null (legacy transactions)
  final String? paymentMethod;

  factory GetTransactionResponseDto.fromJson(Map<String, dynamic> json) =>
      _$GetTransactionResponseDtoFromJson(json);
}

File: apps/api/FinanceControl.Shared/Dtos/Response/GetTransactionResponseDto.cs
using FinanceControl.Shared.Enums;

namespace FinanceControl.Shared.Dtos.Response
{
    public class GetTransactionResponseDto
    {
        public int Id { get; set; }
        public int? BudgetId { get; set; }
        public string? BudgetName { get; set; }
        public int SubCategoryId { get; set; }
        public string SubCategoryName { get; set; }
        public string? SubCategoryEmoji { get; set; }

        /// <summary>
        /// The parent category of <see cref="SubCategoryName"/>. Carried on the row so a
        /// consumer that only has the transaction — the CSV export, for one — can name it
        /// without loading the whole category tree.
        /// </summary>
        public string CategoryName { get; set; } = string.Empty;

        public int AccountId { get; set; }
        public string AccountName { get; set; }
        public int? DestinationAccountId { get; set; }
        public string? DestinationAccountName { get; set; }
        public int? RecurringTransactionId { get; set; }
        public int? ParentTransactionId { get; set; }
        public int Value { get; set; }
        public EnumTransactionType Type { get; set; }
        public string Description { get; set; }
        public DateOnly TransactionDate { get; set; }
        public EnumPaymentType PaymentType { get; set; }
        public EnumPaymentMethod? PaymentMethod { get; set; }
        public int? InstallmentNumber { get; set; }
        public int? TotalInstallments { get; set; }
        public int? AreaId { get; set; }
        public string? AreaName { get; set; }
        public List<GetTagResponseDto> Tags { get; set; } = new();
    }
}
